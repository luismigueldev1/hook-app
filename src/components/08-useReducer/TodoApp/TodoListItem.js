import React from "react";
import PropTypes from "prop-types";

function TodoListItem({ todo, handleDelete, handleToogle }) {
  return (
    <li className="list-group-item d-flex align-items-center justify-content-between">
      <p className="m-0">{todo.done ? <del>{todo.task}</del> : todo.task}</p>
      <div className="buttons">
        <button
          className={`btn btn-sm  ${
            todo.done ? " btn-warning" : " btn-primary"
          }`}
          onClick={() => handleToogle(todo.id)}
        >
          {todo.done}
          {todo.done ? "Pending" : "Done"}
        </button>

        <button
          className="btn btn-sm btn-danger ml-2"
          onClick={() => handleDelete(todo.id)}
        >
          Delete
        </button>
      </div>
    </li>
  );
}

TodoListItem.propTypes = {
  todo: PropTypes.object.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToogle: PropTypes.func.isRequired,
};

export default TodoListItem;
