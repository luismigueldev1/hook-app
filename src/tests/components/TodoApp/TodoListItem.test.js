import { shallow } from "enzyme";
import TodoListItem from "../../../components/08-useReducer/TodoApp/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Pruebas en el componente <TodoListItem/>", () => {
  const handleDelete = jest.fn();
  const handleToogle = jest.fn();
  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      handleDelete={handleDelete}
      handleToogle={handleToogle}
    />
  );

  test("debe de renderizarze correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de llamar la funcion handleDelete", () => {
    wrapper.find(".btn-danger").simulate("click");

    expect(handleDelete).toHaveBeenCalledWith(1);
  });

  test("debe de llamar la funcion handleToogle", () => {
    wrapper.find(".btn-primary").simulate("click");
    expect(handleToogle).toHaveBeenCalled();
  });

  test("debe de mostrar el text correctamente", () => {
    const p = wrapper.find("p").text().trim();
    expect(p).toBe(demoTodos[0].task);
  });
});
