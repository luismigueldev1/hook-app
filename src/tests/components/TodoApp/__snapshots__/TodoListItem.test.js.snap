// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <TodoListItem/> debe de renderizarze correctamente 1`] = `
<li
  className="list-group-item d-flex align-items-center justify-content-between"
>
  <p
    className="m-0"
  >
    Aprender React
  </p>
  <div
    className="buttons"
  >
    <button
      className="btn btn-sm   btn-primary"
      onClick={[Function]}
    >
      Done
    </button>
    <button
      className="btn btn-sm btn-danger ml-2"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</li>
`;
